### DISCUSSION TITLE
Why does Recast/Detour use its own array and vector implementations instead of stl?

### DISCUSSION THREAD

**Discussion by liubai01** [recastnavigation/recastnavigation#551] [Q&A] [answered]
For example, in `RecastCounter.cpp`:
```cpp
...
rcIntArray verts(256);
rcIntArray simplified(64);

for (int y = 0; y < h; ++y)
{
...
```
Why not use `std::vector` to replace it? Is it because when recast navigation born, the stl containers have not born(similar reason as `TArray` in ue4)?

**Comment by grahamboree** [comment, marked as answer]
`std::vector` long precedes recast.  Many games avoid using the STL for various reasons, so for maximum compatibility Recast also avoids using STL containers.
